compiler/rustc_mir_build/src/build/coverageinfo.rs

```diff
@@ -364,13 +385,17 @@ impl Builder<'_, '_> {
                 |block| branch_info.inject_block_marker(&mut self.cfg, source_info, block);
             let true_marker = inject_block_marker(then_block);
             let false_marker = inject_block_marker(else_block);
-            let condition_info =
-                branch_info.fetch_mcdc_condition_info(self.tcx, true_marker, false_marker);
+            let (decision_depth, condition_info) = branch_info
+                .fetch_mcdc_condition_info(self.tcx, true_marker, false_marker)
+                .map_or((0, None), |(decision_depth, condition_info)| {
+                    (decision_depth, Some(condition_info))
+                });
```

Comment:
I wonder if it makes more sense to inline this special case into the end of `fetch_mcdc_condition_info`, and have that function return `(u16, Option<ConditionInfo>)` instead.

Reply:
This can be fixed at #124399 ,where `BranchInfoBuilder` does not call `fetch_mcdc_condition_info` any more.

Reply:
@ZhuUx If you're happy to clean this up in #124399, then I don't mind leaving it a little messy in this PR. 👍